### ISSUE TITLE
screen reader issue since 1.89.0 with open file CTRL+p

### ISSUE THREAD

**Issue by dstpierre** [microsoft/vscode#211976]
Does this issue occur when all extensions are disabled?: Yes/No
Yes

- VS Code Version: 1.89.0
- OS Version: Windows 10

Steps to Reproduce:

I'm using NVDA as screen reader, before 1.89.0 this worked:

1. Opening the "search file by name or symbol" with `CTRL+p`
2. If I type path where only 1 file exists, the file name isn't "announced", cannot do up/down arrow as there's only one result.
3. In fact, the first file isn't announce for result with more than one results.

To be clear, image we have the following file/directory tree

|- dir1/
|--- abc.go
|- main.go

Doing CTRL+p and than typing `dir1/` it displayed the `abc.go` file but screen reader says nothing.

For result with multiple file, the first file of the list isn't "sent" to screen reader, doing down arrow and the second file in list is "said" by scree reader.

If I may this is a huge problem.
Thanks

**Comment by TylerLeonhardt** [comment]
Looking at NVDA's log, I see this:
```
d
1 Results  
Search files by name (append : to go to line or @ to go to symbol)  list
abc.go dir1, file results  1 of 1
i
r
slash
```
with your example:
![image](https://github.com/microsoft/vscode/assets/2644648/8faa0fe5-cf91-4b01-b4f4-92dbbf109f2c)
```

So we _do_ announce the name, but the second you type another character that gets lost. In 1.88 it would re-say the file.

**Comment by TylerLeonhardt** [comment]
The change that caused this regression is here:
https://github.com/microsoft/vscode/pull/208913

cc @joaomoreno

**Comment by TylerLeonhardt** [comment]
and locally, in main, I reverted that commit and the issue went away.

It looks like we were depending on that insert in order to trigger the a11y event to read things out.

Repro:
* I used NVDA on Windows
* have a workspace with the following structure:
  * `dir/`
    * `abc.go`
  * `main.go`
* Open the file picker (ctrl+P)
* type, slowly, `d` `I` `r`.
* :bug: file abc.go only gets read out on the d and not the other typed characters

**Comment by joaomoreno** [comment]
The tree seems to be doing the right thing for performance sake: it is not modifying the DOM. Before #208913, the list row was thrown away and rendered again for no good reason. That had the nice side effect of A11Y being read as we desired.

Trying to find a way to _kick_ NVDA to still read it, even though nothing has changed.

**Comment by alexr00** [comment]
I think I am still reproing the original issue:
- I made the folder structure from https://github.com/microsoft/vscode/issues/211976#issuecomment-2093805691
- `ctrl+p`
- Type `d`, then before it can announce the name of the file, type `i`
- `abc.go` is never announced.

**Comment by alexr00** [comment]
I didn't have screen reader accessibility mode turned on.

**Comment by dstpierre** [comment]
This issue is back in version `1.93.1` might have been the version prior to that as well. /cc @alexr00

**Comment by TylerLeonhardt** [comment]
I'll reopen for now and let @joaomoreno decide next steps.

**Comment by dstpierre** [comment]
@dmitrivMS yes it's working with the version I have, which is 1.105.1.

Thanks I'll close the issue.
